src/plugins/telemetry/server/fetcher.ts

```diff
@@ -94,6 +97,10 @@ export class FetcherTask {
     }
   }
 
+  private async areAllCollectorsReady() {
+    return await this.telemetryCollectionManager!.areAllCollectorsReady();
```

Comment:
Super-NIT: Maybe it's safer to count on the possibility of `this.telemetryCollectionManager` to be undefined (just in case) 🙂 

```suggestion
    return (await this.telemetryCollectionManager?.areAllCollectorsReady()) ?? false;
```